aws/data_source_aws_outposts_outpost_instance_type.go

```diff
@@ -0,0 +1,128 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/outposts"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func dataSourceAwsOutpostsOutpostInstanceType() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsOutpostsOutpostInstanceTypeRead,
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
Should this also call `validateArn()`? Also seems weird that `validateArn()` allows an empty string.

Reply:
Will swap this and re-verify 👍